### DISCUSSION TITLE
Popover mouseOver

### DISCUSSION THREAD

**Discussion by chjpark** [twilio-labs/paste#807] [Q&A] [answered]
Does the Popover allow override to the onClick event to show the Popover? I'd like to be able to show a Popover when I over over an element that is associated with a Popover.

**Comment by serifluous** [comment, marked as answer]
Hey @chjpark ! Can you share (screenshot, codesandbox, etc) what you're trying to use to trigger the popover, and what content is in the popover?

Since Popover is a type of dialog, it must be triggered by an explicit user action like a click, and not hover.

If you need something similar to show up on hover, you can use a [Tooltip](https://paste.twilio.design/components/tooltip/). However Tooltips can have only text in them, so depending on your use case, that might not work out for you.

**Comment by elainelylee** [comment]
How can we compose something like a tooltip but allows us to show more information like a popover? For example, 
<img width="388" alt="Screen Shot 2020-11-16 at 5 47 32 PM" src="https://user-images.githubusercontent.com/6718764/99329795-d6602380-2833-11eb-96cd-1418e4fae6b7.png">

**Reply by elainelylee** [reply]
Do we need to change the text to a clickable thing? 
![image](https://user-images.githubusercontent.com/6718764/99330030-5b4b3d00-2834-11eb-9a9c-f4019b510e01.png)

I added an underline here for example. It's not an anchor nor button. Or, maybe it's a button but doesn't take in the blue text style.

**Reply by elainelylee** [reply]
For example, popover shows up after click. The popover potentially will have an avatar if that's configured by the user. (That's the idea)

![image](https://user-images.githubusercontent.com/6718764/99331449-c563e200-2834-11eb-853f-c89840584c17.png)

**Reply by SiTaggart** [reply]
A Popover must be triggered by a dedicated button. We can't show a non-modal dialog on hover / focus as they steal a users curser focus, so a dedicated user action must be taken before we start moving you around the screen. 

The button trigger can be any variant of our button; Primary, secondary, destructive, link style, icon, icon only.

The button trigger must be easily distinguishable as actionable (different) from the text and other elements surrounding it, including underlined text.

**Reply by elainelylee** [reply]
Would having the underline added to the text work? Referring to the snapshot above where on hover the text "qualified2387" is underlined

**Reply by kate-porter** [reply]
Are you able to use a link-style button instead? Or is there a specific reason you are hoping to use black text for this?

**Reply by SiTaggart** [reply]
Unfortunately using something like underlined text wouldn't make much difference in helping identify it as actionable. If there was some underlined text on your screen, the actionable one would look identical to the non-actionable one.

We can't really take cases like this on a page by page basis either. You might say that there isn't any underlined text on my screen, so it _is_ visually different from non-actionable text, but we would be treating the page as an isolated experience, and not one that lives within a much larger app. Grey underline text everywhere in the app is just that, grey underlined text. It isn't actionable.

If a user who is used to that pattern else where in one of our apps visits your page, they are likely to not know it's a button as we've trained them to know underlined grey text, is just underlined text.

We sometimes do have to make design decisions based on how how they fit into the whole app experience and not just on a page by page basis unfortunately.

**Reply by elainelylee** [reply]
I meant, the underline shows up on hover.
We tested out the link-style button. The reset version doesn't have the underline on it on hover, from what we saw in Paste. Was wondering if there should be an underline there on hover and if that's something we can use.

The text is in a table, and we're thinking of giving more information to the user when they hover/click on that data piece. It could be a name or an ID that the developer configured in the table.

**Reply by kate-porter** [reply]
As Simon called out, a button trigger needs to be easily distinguishable from surrounding static text prior to any interaction (like hover), so it is clear to all users that the text is actionable. Black text that gets an underline on hover would not accomplish this, as it would look just like the surrounding text.

My recommendation for this would be to use a popover (you can put whatever you want in a popover, so it would support the addition of an avatar), and have the "qualified2387" text a link-style button that triggers it on click. You could also use an icon-only link-style button _next to_ the "qualified2387" text that triggers the popover on click, instead of triggering it with the text.

**Reply by elainelylee** [reply]
![image](https://user-images.githubusercontent.com/6718764/99458744-e1718d00-28e1-11eb-83a2-35d28250b5c7.png)

Here's an example of having the icon button. 

Can also try using the blue link-button, but be a little unexpected the first time users interact with it thinking it will bring them somewhere but just opens a small popover. For the case where we have "ID unavailable" as a placeholder, we may not be able to have a blue text-link there as it will no longer look like a placeholder. What are your thoughts?

![image](https://user-images.githubusercontent.com/6718764/99459477-04e90780-28e3-11eb-9b91-590c342f1b35.png)
In the cases where there are avatars, we would be able to make that the button. But, that's not always the case.

A different but related question that I can't find the answer to. Do we allow line breaks in the tool tip? If so, we may be able to use that and not include an avatar.

**Reply by SiTaggart** [reply]
I'm not sure I understand what question you're asking for the first one, but for the second:

> Do we allow line breaks in the tool tip?

Tooltips should only be single line, short string with no formatting. The API of the components also only accepts that. The reason being, is that because tooltips are supplementary content, there is no way to communicate formatting to assistive technology. A screen reader for example only announces the content as a single string and removes all grammar and formatting, concatenating some words together, so the text is no longer understandable. As it is supplementary a screen reader also doesn't provide the usual text inspect controls you'd find in normal screen text. For example, a long complex paragraph can be announced word by word, letter by letter, go backwards or forwards, to make it easier to understand. None of those controls are provided for supplementary text so if the content is long and complex, it's very hard for a screen reader to understand it.

**Reply by elainelylee** [reply]
![image](https://user-images.githubusercontent.com/6718764/99599260-d5540100-29af-11eb-9943-3103031abd7b.png)
![image](https://user-images.githubusercontent.com/6718764/99599283-e3098680-29af-11eb-833b-418bedefd435.png)

This is the latest version. User clicks on link button to get the popover. And, user clicks on the info icon to get to popover.

**Reply by kate-porter** [reply]
Just a followup from office hours - our recommendation on this would be to use the same style for everything in the table that performs the same action. So, instead of using the info icon to trigger the popover when name/ID/etc. are unavailable, make the "name unavailable" or "ID unavailable" text a link-style button that triggers the popover, and forgo the info icon.
